Analysis chapter

kain88-de opened the pull request:
This is a start for the analysis module section. A little bit of work left but it would be nice to hear if this goes into the right direction.

richardjgowers commented on papers/oliver_beckstein/oliver_beckstein.rst:
```diff
@@ -90,9 +90,31 @@ The MDAnalysis library :cite:`Michaud-Agrawal:2011fu` addresses this problem by
 
+In the MDAnalysis.analysis module we provide a large variety of standard analysis algorithms, like RMSD, alignment, native contacts, as well as unique algorithms, like the LeaftleftFinder and Path Similarty Analysis. We have recently started to unify the interface to the different algorithms with an `AnalysisBase` class. Currently PersistenceLength, InterRDF, LinearDensity and Concats analysis ha ...
+
+The new Contacts class is a good example a generic API that allows easy adaptations of algorithms while still offering an easy setup for standard analysis types. For contact analysis it is possible to have different definitions based on coordinates alone **Cite two papers from docs**. We have designed the API to choose between common definitions and pass used defined functions to develop new defi ...
```
I'd make a bigger deal that we can inject kernels into generic analysis which makes it more flexible than other tools that are essentially black boxes

kain88-de replied:
Is kernels the right word for this? So far I have used definitions in the text but I'm not super happy with that

richardjgowers replied:
Maybe kernel is wrong.. metric?

kain88-de commented:
@orbeckst do you remember where the q1q2 analysis first showed up? I only know the original MDAnalysis paper

kain88-de commented:
Included @richardjgowers suggestions

orbeckst commented:
> On 24 May, 2016, at 08:24, kain88-de notifications@github.com wrote:
> 
> @orbeckst do you remember where the q1q2 analysis first showed up? I only know the original MDAnalysis paper

I got it from Mark Delarue’s MinActionPath paper [1 and they don’t reference anyone else for it]. I think I also collected a few references in the issue report for the remodelling of the contacts analysis class – didn’t you add them to the docs anyway?

[1] J. Franklin, P. Koehl, S. Doniach, and M. Delarue. MinActionPath: maximum likelihood trajectory for large- scale structural transitions in a coarse-grained locally harmonic energy landscape. Nucleic Acids Res., 35 (Web Server issue):W477–W482, 2007. doi: 10.1093/nar/gkm342. URL http://nar.oxfordjournals. org/cgi/content/abstract/35/suppl_2/W477.

## 

Oliver Beckstein \* orbeckst@gmx.net
skype: orbeckst  \* orbeckst@gmail.com

kain88-de commented:
That is the paper I added and as far as I remember the only one you listed in the PR. But I'll have a look again.

orbeckst commented on papers/oliver_beckstein/oliver_beckstein.rst:
```diff
@@ -90,9 +90,28 @@ The MDAnalysis library :cite:`Michaud-Agrawal:2011fu` addresses this problem by
+Analysis Module
+---------------
 
+In the MDAnalysis.analysis module we provide a large variety of standard analysis algorithms, like RMSD, alignment, native contacts, as well as unique algorithms, like the LeaftleftFinder and Path Similarty Analysis. We have recently started to unify the interface to the different algorithms with an `AnalysisBase` class. Currently PersistenceLength, InterRDF, LinearDensity and Contacts analysis h ...
```
reference the old MDA paper for LeafletFinder

orbeckst replied:
We should probably add mathematical definitions for these tools.

kain88-de replied:
Definition for what Tools exactly?

orbeckst replied:
> On 25 May, 2016, at 09:51, kain88-de notifications@github.com wrote:
> 
> Definition for what Tools exactly?

For the Comp Sci people, write down the equations that need to be solved, e.g., for the RMSD, RMSF, R_gyr, …. They like to be able to judge the complexity of the task. The audience is not just the biophysics community so we have to explain some things that are trivial to us. Furthermore, it would be good to have all the formal definitions in one place. 

Just add what you want/can contribute for the moment and leave gaps elsewhere.

kain88-de replied:
Seriously? That would take a paper in itself. RMSD is pretty easy to define but radius of gyration, alignment those are harder with out context. The path similarity even has it's own paper. I'm happy to cite books for the tasks that give proper context but writing out all formulas for the analysis modules is a bit much.

For the native contacts I can understand this as I go into more detail for that module.

orbeckst commented on papers/oliver_beckstein/oliver_beckstein.rst:
```diff
@@ -90,9 +90,28 @@ The MDAnalysis library :cite:`Michaud-Agrawal:2011fu` addresses this problem by
+In the MDAnalysis.analysis module we provide a large variety of standard analysis algorithms, like RMSD, alignment, native contacts, as well as unique algorithms, like the LeaftleftFinder and Path Similarty Analysis. We have recently started to unify the interface to the different algorithms with an `AnalysisBase` class. Currently PersistenceLength, InterRDF, LinearDensity and Contacts analysis h ...
+
+The new Contacts class is a good example a generic API that allows easy adaptations of algorithms while still offering an easy setup for standard analysis types. For contact analysis it is possible to have different metrics based on coordinates alone :cite:`Best2013,Franklin2007`. We have designed the API to choose between common metrics and pass user defined functions to develop new metrics. Thi ...
+
```
no links to docs, they are just too volatile. We should have a section somewhere that says where docs can be found http://docs.mdanalysis.org

orbeckst replied:
... a section with a link to the docs already exists.

Generally speaking, I find that papers, as the fairly permanent record, are the place to explain algorithms and "unchanging" ideas so I rather put it in a paper to start with and then reference the paper.

orbeckst commented on papers/oliver_beckstein/oliver_beckstein.rst:
```diff
@@ -90,9 +90,28 @@ The MDAnalysis library :cite:`Michaud-Agrawal:2011fu` addresses this problem by
+
+The new Contacts class is a good example a generic API that allows easy adaptations of algorithms while still offering an easy setup for standard analysis types. For contact analysis it is possible to have different metrics based on coordinates alone :cite:`Best2013,Franklin2007`. We have designed the API to choose between common metrics and pass user defined functions to develop new metrics. Thi ...
+
+.. code-block:: python
```
code examples are great!

orbeckst commented:
Fine with me, I'd merge but you need to fix the conflicts first.

kain88-de committed:
Add analysis chapter

kain88-de committed:
Add paper citations

kain88-de commented:
I rebased this against the 2016 branch. I feel like I should add bit more information though, I think the text is still hard to understand for someone who isn't a coredev.

kain88-de committed:
Add references for used analysis algorithms

They are the same references we cite in the docs

kain88-de committed:
add note why API needs to be unified

kain88-de committed:
go into detail for contacts analysis

kain88-de commented:
@orbeckst 

I'm happy now with the first paragraph. I added a note why we need to unify the API and added citations for all used analysis algorithms from the docs if specified.

For the second paragraph I'm going into to more detail what the contacts class does and what a native contact is. That is already better then previously but I think could be improved still. 

I'm good to have this merged. I can still make small PR's if I want more changes.

orbeckst commented on papers/oliver_beckstein/oliver_beckstein.rst:
```diff
@@ -116,6 +116,40 @@ If you like to use MDAnalysis for your project please join our community_ board.
+Analysis Module
+---------------
+
+In the MDAnalysis.analysis module we provide a large variety of standard analysis algorithms, like RMSD, alignment :cite:`PuLiu_FastRMSD_2010`, native contacts :cite:`Best2013,Franklin2007`, as well as unique algorithms, like the LeaftleftFinder :cite:`Michaud-Agrawal:2011fu` and Path Similarty Analysis :cite:`Seyler:2015fk`.
```
typo: Similarity

orbeckst commented on papers/oliver_beckstein/oliver_beckstein.rst:
```diff
@@ -116,6 +116,40 @@ If you like to use MDAnalysis for your project please join our community_ board.
+---------------
+
+In the MDAnalysis.analysis module we provide a large variety of standard analysis algorithms, like RMSD, alignment :cite:`PuLiu_FastRMSD_2010`, native contacts :cite:`Best2013,Franklin2007`, as well as unique algorithms, like the LeaftleftFinder :cite:`Michaud-Agrawal:2011fu` and Path Similarty Analysis :cite:`Seyler:2015fk`.
+Historically the algorithms have been contributed by various researchers to satisfy their own needs, this lead to some fragmentation in the interface of different algorithms.
```
Historically these algorithms were contributed by various researchers as individual modules to satisfy their own needs but this lead to some fragmentation in the user interface to these modules.

orbeckst commented on papers/oliver_beckstein/oliver_beckstein.rst:
```diff
@@ -116,6 +116,40 @@ If you like to use MDAnalysis for your project please join our community_ board.
+We have recently started to unify the interface to the different algorithms with an `AnalysisBase` class.
+Currently PersistenceLength, InterRDF, LinearDensity and Contacts analysis have been ported.
+If applicable we also strive to make the API's to the algorithms generic.
+Most other tools hand the user analysis algorithms as black boxes, we want to avoid that and give the users all he needs to adapt an analysis to his/her needs.
```
Most other tools hand the user analysis algorithms as black boxes. We ...

(Do not join two independent full clauses with a comma. Use periods or conjunctions.)

orbeckst commented on papers/oliver_beckstein/oliver_beckstein.rst:
```diff
@@ -116,6 +116,40 @@ If you like to use MDAnalysis for your project please join our community_ board.
+In the MDAnalysis.analysis module we provide a large variety of standard analysis algorithms, like RMSD, alignment :cite:`PuLiu_FastRMSD_2010`, native contacts :cite:`Best2013,Franklin2007`, as well as unique algorithms, like the LeaftleftFinder :cite:`Michaud-Agrawal:2011fu` and Path Similarty Analysis :cite:`Seyler:2015fk`.
+Historically the algorithms have been contributed by various researchers to satisfy their own needs, this lead to some fragmentation in the interface of different algorithms.
+We have recently started to unify the interface to the different algorithms with an `AnalysisBase` class.
+Currently PersistenceLength, InterRDF, LinearDensity and Contacts analysis have been ported.
```
Currently, the PersistenceLength, InterRDF, LinearDensity and Contacts analysis have been ported to the unified user interface. PersistenceLength calculates the persistence length of a polymer, InterRDF calculates XXX, LinearDensity generates a density along a given axis, and Contacts analysis native contacts, as described in more detail below.

orbeckst commented on papers/oliver_beckstein/oliver_beckstein.rst:
```diff
@@ -116,6 +116,40 @@ If you like to use MDAnalysis for your project please join our community_ board.
+A common quantity of interest is the fraction of native contacts, native contacts are all atoms that are nearby in the reference.
+For native contacts there exists two metrics :cite:`Best2013,Franklin2007` and we default to the later.
+We have designed the API to choose between the two metrics and pass user defined functions to develop new metrics or measure other quantities.
+This generic interface allowed us to implement a q1q2 analysis :cite:`Franklin2007` ontop of the Contacts class.
```
on top

orbeckst commented:
Only small changes, see above.

orbeckst commented:
It would be good to have references for the analysis algorithms or quantitites that are calculated — can be text books, e.g., for persistence length.

kain88-de committed:
Add some algorithm describtions & small changes

kain88-de commented:
I included suggestions. I don't have good references for the persistence length or RDF's. @richardjgowers implemented those and should know some good values (I think)
